### ISSUE TITLE
iOS app not detected

### ISSUE THREAD

**Issue by nidhoeggr09** [sarriaroman/FabricPlugin#56]
I am working on a cordova app for android and ios. I added you plugin as described. The android version was detected instantly and everything seems to work fine. However i don't see the ios version of my app in the fabric dashboard. My API keys seem to be working fine - otherwise the android app wouldn't work. I also tried to remove and add the platform as well as the plugin. Calling `window.fabric.Crashlytics.sendCrash();` actually crashes the app, so i think the plugin was integrated successfully. Could it be a problem that i am deploying with Visual Studio 2015 on a remote mac?

**Comment by Justin-Credible** [comment]
The `platforms/ios` directory must exist before the plugin is added (`cordova platform add ios` needs to be executed before `cordova plugin cordova-fabric-plugin`). This is because a build hook runs on `after_plugin_install` to update the XCode (or Android) projects.

**Comment by nidhoeggr09** [comment]
Yes, this was the case. Do you have any other ideas what could have gone wrong?

**Comment by Justin-Credible** [comment]
You can open the iOS project in XCode and verify that the build hook added the "Fabric.io Crashlytics" build phase script.

**Comment by nidhoeggr09** [comment]
Looks good so far. Can it be a problem with the dashboard? Do i  have to do anything there? Always when i try to click "Add app" it detects my Android version and displays it. But that shouldnt be a thing, right?

![screen shot 2016-08-05 at 00 05 28](https://cloud.githubusercontent.com/assets/1476386/17420058/eca8fd20-5aa0-11e6-96b2-dd2054bd3266.png)

**Comment by Justin-Credible** [comment]
At this point you should be able to launch via XCode on a device or the
simulator and your app should show up in Fabric. If it still isn't showing
up you should look in the XCode console to see if there are any errors.

**Comment by nidhoeggr09** [comment]
You were right, there is an error! Now i copied the whole VS project folder to my virtual machine and tried to build the project in XCode. I'll get the following error (although it never showed up when doing the remote build in  Visual Studio on my PC):

`/Users/thomas/Library/Developer/Xcode/DerivedData/MyProjectXY-hgdrjzjfgbqnjsczzjvhlrjmjkif/Build/Intermediates/MyProjectXY.build/Debug-iphonesimulator/MyProjectXY.build/Script-9A2435C036BF40FFA9766920.sh: line 2: ../../plugins/cordova-fabric-plugin/lib/ios/Crashlytics.framework/run: Permission denied`

Now the first thing i would do is to try to get it run on the mac. I am not so familiar with OSX - where and of what would i change the permission?

The second step would be to get it run via the remote build in Visual Studio. Do you have and experiences with it? I think since Visual Studio 2015 Community, many  people are using it like i do to create Cordova apps ([install-vs-tools-apache-cordova](http://taco.visualstudio.com/en-us/docs/install-vs-tools-apache-cordova/#ios), [taco.tools](http://taco.tools/docs/remote-build.html)).

**Comment by Justin-Credible** [comment]
It looks like the run binary doesn't have execute permission, most likely because the directory was copied over from an NTFS file system. You could `chmod +x` it, but it would probably be best to remove the platforms and plugins directories, re-add the platforms, and re-add the plugins on the OS X machine.

**Comment by nidhoeggr09** [comment]
Sorry for my late response, but my dog had a fight with my smartphone and unfortunately he won :) 

I managed to get it work with the steps you said:

1) Copy project folder to virtual machine
2) Remove platform ios
3) Remove fabric plugin
4) Add platform ios
5) Add fabric plugin
6) Prepare ios
7) Open and build in XCode

It's not an ideal solution since i have to do this every time i want to build a release version. But for now it should be sufficient. Thank you for be so patient with me!

**Comment by CelsoSantos** [comment]
I don't think the after_plugin_install hook is working.

I've tried everything in this thread and still I don't get that build phase when on Xcode, neither do I see information that that hook is being run upon installing the plugin:

```
ionic plugin add cordova-fabric-plugin --save --variable FABRIC_API_KEY="mykey" --variable FABRIC_API_SECRET="mysecret"
Fetching plugin "cordova-fabric-plugin" via npm

Installing "cordova-fabric-plugin" for android
Installing "cordova-fabric-plugin" for ios
Saved plugin info for "cordova-fabric-plugin" to config.xml

Running command: /Users/dev/Syone/myloc/myloc-ionic/hooks/after_plugin_add/010_register_plugin.js /Users/dev/Syone/brandbassador/brandbassador-ionic
```

XCode version: 8.1 (8B62)
Node version: 6.7.0
NPM version: 3.10.3

The output of `ionic info`:
```
Your system information:

Cordova CLI: 6.3.1
Gulp version:  CLI version 3.9.1
Gulp local:
Ionic CLI Version: 2.1.0
Ionic App Lib Version: 2.1.0-beta.1
ios-deploy version: 1.9.0
ios-sim version: 5.0.8
OS: Mac OS X El Capitan
Node Version: v6.7.0
Xcode version: Xcode 8.1 Build version 8B62
```

Can anyone help me? I'm facing the same issue. The app crashes upon start. Everything works fine in Android, only iOS giving me trouble.

Best Regards,
Celso Santos
